aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp

```diff
@@ -204,14 +212,24 @@ namespace Aws
                 ByteBuffer digest(length);
                 memset(digest.GetUnderlyingData(), 0, length);
 
-                HMAC_CTX ctx;
-                HMAC_CTX_init(&ctx);
+                HMAC_CTX *ctx;
+#if OPENSSL_VERSION_NUMBER < 0x10100003L
+                HMAC_CTX _ctx;
+                ctx = &_ctx;
+                HMAC_CTX_init(ctx);
+#else
+                ctx=HMAC_CTX_new();
```

Comment:
I believe so. Several structures have been made opaque in OpenSSL 1.1, so the OpenSSL allocators have to be used, see https://www.openssl.org/docs/man1.1.0/crypto/HMAC_CTX_free.html unless there is something I missed.